data-prepper-plugins/parse-json-processor/src/main/java/com/amazon/dataprepper/plugins/processor/parsejson/ParseJsonProcessor.java

```diff
@@ -76,9 +88,62 @@ public void shutdown() {
 
     }
 
+    private Map<String, Object> parseUsingPointer(final Event event, final Map<String, Object> parsedJson, final String pointer,
+                                                  final boolean doWriteToRoot) {
+        final Event temporaryEvent = JacksonEvent.builder().withEventType("event").build();
+        temporaryEvent.put(source, parsedJson);
+
+        final String trimmedPointer = trimPointer(pointer);
+        final String actualPointer = source + "/" + trimmedPointer;
+
+        final boolean pointerIsValid = temporaryEvent.containsKey(actualPointer);
+        if (!pointerIsValid) {
+            LOG.error("Writing entire JSON because the pointer {} is invalid on Event {}", pointer, event);
+            return parsedJson;
+        }
+
+        final Object valueAtPointer = temporaryEvent.get(actualPointer, Object.class);
+        final String endOfPointer = getEndOfPointer(trimmedPointer);
+
+        final boolean shouldUseEntirePointerAsKey = event.containsKey(endOfPointer) && doWriteToRoot;
+        if (shouldUseEntirePointerAsKey) {
+            return Collections.singletonMap(normalizePointerStructure(trimmedPointer), valueAtPointer);
+        }
+
+        return Collections.singletonMap(normalizePointerStructure(endOfPointer), valueAtPointer);
+    }
+
+    private String getEndOfPointer(final String trimmedPointer) {
+        final ArrayList<String> elements = new ArrayList<>(Arrays.asList(trimmedPointer.split("/")));
+        if (elements.size() <= 1) return trimmedPointer;
+
+        final boolean lastElementInPathIsAnArrayIndex = elements.get(elements.size()-1).matches("[0-9]+");
+
+        if (lastElementInPathIsAnArrayIndex) {
```

Comment:
Numeric Strings (e.g. "445") are valid as JSON keys but they are not supported by `JacksonEvent`s because within `JacksonEvent` every numeric string is treated as [an array index](https://github.com/opensearch-project/data-prepper/blob/eb94f4abbd4f83ce3b3ebe4d4478af7531f9bc36/data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java#L119).

I don't think that numeric keys are a common use case, but there are some examples of people using them online, such as [here](https://stackoverflow.com/questions/23552708/how-to-parse-json-with-number-as-a-key).

If support for numeric keys is added to `JacksonEvent`, this code could be modified like the following to support pointers to numeric keys: 


```
final String pointerExcludingLastField = source + "/" + elements.subList(0,elements.size()-1)
                    .stream().collect(Collectors.joining("/"));
final Object valueAtPointerExcludingLastField = temporaryEvent.get(pointerExcludingLastField, Object.class);
if (valueAtPointerExcludingLastField instanceof List) {
    final String lastTwoElements = elements.get(elements.size()-2) + "/" + elements.get(elements.size()-1);
    return lastTwoElements;
}
```